frontend/server/integration-tests/artifact-auth.test.ts

```diff
@@ -0,0 +1,364 @@
+        MINIO_PORT: '9000',
+        MINIO_SECRET_KEY: 'minio123',
+        MINIO_SSL: 'false',
+        ML_PIPELINE_SERVICE_HOST: 'localhost',
+        ML_PIPELINE_SERVICE_PORT: '8888',
+        KUBEFLOW_USERID_HEADER: 'kubeflow-userid',
+        KUBEFLOW_USERID_PREFIX: '',
+      });
+
+      configurations.auth.enabled = true;
+
+      app = new UIServer(configurations);
+
+      const request = requests(app.start());
+      request
+        .get(
+          '/artifacts/get?source=minio&bucket=ml-pipeline&key=hello%2Fworld.txt&namespace=../../../etc',
+        )
+        .set('kubeflow-userid', 'user@example.com')
+        .expect(400)
+        .expect(response => {
+          expect(response.text).toContain('Invalid namespace');
+        })
+        .end(done);
+    });
+
+    it('rejects unauthorized cross-namespace access', done => {
+      mockAuthorize.mockRejectedValue({
+        status: 403,
+        statusText: 'Forbidden',
+        text: () => Promise.resolve('User is not authorized'),
+      });
+
+      const configurations = loadConfigs(argv, {
+        MINIO_ACCESS_KEY: 'minio',
+        MINIO_HOST: 'minio-service',
+        MINIO_NAMESPACE: 'kubeflow',
+        MINIO_PORT: '9000',
+        MINIO_SECRET_KEY: 'minio123',
+        MINIO_SSL: 'false',
+        ML_PIPELINE_SERVICE_HOST: 'localhost',
+        ML_PIPELINE_SERVICE_PORT: '8888',
+        KUBEFLOW_USERID_HEADER: 'kubeflow-userid',
+        KUBEFLOW_USERID_PREFIX: '',
+      });
+
+      configurations.auth.enabled = true;
+
+      app = new UIServer(configurations);
+
+      const request = requests(app.start());
+      request
+        .get(
+          '/artifacts/get?source=minio&bucket=ml-pipeline&key=hello%2Fworld.txt&namespace=other-namespace',
+        )
+        .set('kubeflow-userid', 'user@example.com')
+        .expect(403)
+        .expect(response => {
+          expect(response.text).toContain('not authorized');
+        })
+        .end(done);
+    });
+
+    it('allows authorized namespace access', done => {
+      mockAuthorize.mockImplementation(() =>
+        Promise.resolve({
+          ok: true,
+          status: 200,
+          json: () => Promise.resolve({}),
+          text: () => Promise.resolve(''),
+        }),
+      );
+
+      const configurations = loadConfigs(argv, {
+        MINIO_ACCESS_KEY: 'minio',
+        MINIO_HOST: 'minio-service',
+        MINIO_NAMESPACE: 'kubeflow',
+        MINIO_PORT: '9000',
+        MINIO_SECRET_KEY: 'minio123',
+        MINIO_SSL: 'false',
+        ML_PIPELINE_SERVICE_HOST: 'localhost',
+        ML_PIPELINE_SERVICE_PORT: '8888',
+        KUBEFLOW_USERID_HEADER: 'kubeflow-userid',
+        KUBEFLOW_USERID_PREFIX: '',
+      });
+
+      configurations.auth.enabled = true;
+
+      app = new UIServer(configurations);
+
+      const request = requests(app.start());
+      request
+        .get(
+          '/artifacts/get?source=minio&bucket=ml-pipeline&key=hello%2Fworld.txt&namespace=my-namespace',
```

Comment:
Where does my-namespace come from? Where is definded that user@example.com has access to my-namespace?